### PR TITLE
[BE] Package dynamic_dims/constraint_dims into CreateSymbolicPolicy

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113802

This will make it more convenient to propagate more information through
all of these functions in the future (e.g., for storage offset
information.)

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_subclasses/fake_tensor.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 if TYPE_CHECKING:
     # Import the following modules during type checking to enable code intelligence features
     # Do not import unconditionally, as they import sympy and importing sympy is very slow
-    from torch.fx.experimental.symbolic_shapes import DimConstraint, DimDynamic
+    pass
```
Comment (davidberard98) [inline torch/_subclasses/fake_tensor.py:55]: we can probably just delete this block

**Review by davidberard98 — APPROVED** [review]
apparently some other tests need refactoring as well, otherwise lgtm!

**Review by voznesenskym — APPROVED** [review]

**Pull request closed without merging**